socrates/lib/subscribers/subscriber.js

```diff
@@ -36,7 +36,15 @@ class Subscriber {
   }
 
   livesInGermany() {
-    return this.country() === 'DE';
+    return this.state.country && this.state.country.length > 0 && this.country() === 'DE';
```

Comment:
Wenn Du sicher gehen willst, dass das Object immer sinnvoll befüllt ist, sollte die Domänenlogik das tun. In dem Fall die addon() Funktion, bzw. der Konstruktor. Oder man schützt die Funktion durch ein Guard.